.github/workflows/ci.yml

```diff
@@ -2,13 +2,17 @@ name: build
 
 on:
   pull_request:
+  push:
```

Comment:
I think `pull_request` would be enough.

Whenever there is a new commit(s) being pushed to the HEAD of the PR, [a `synchronize` event](https://github.com/orgs/community/discussions/24567#discussioncomment-3244510) will be dispatched and also trigger this workflow ([ref](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request)).

Reply:
Weird, when I pushed 6f7d53b the CI didn't run, and I added this temporarily to fix the issue and run the CI for every push, but by looking at the list of commit it seems that the CI did run for f257955 even before I added the `push` trigger in c5462a9, so you are probably right.